### PR TITLE
Mesos HA with improved zookeeper

### PR DESCRIPTION
**Pull request by LLParse**

### CODE REVIEW

File: resources/content/config-content/system-stacks/mesos/docker-compose.yml [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
+  net: none
+  volumes:
+  - /data
+  - /log
 mesos-master:
   image: rancher/mesos-master:latest
   environment:
     MESOS_CLUSTER: Rancher
-    MESOS_QUORUM: 1
+    MESOS_QUORUM: 2
```
Comment (LLParse) [inline resources/content/config-content/system-stacks/mesos/docker-compose.yml]: HOST_FAILURE_TOLERANCE + 1

File: resources/content/config-content/system-stacks/mesos/rancher-compose.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 zookeeper:
+  scale: 3
```
Comment (LLParse) [inline resources/content/config-content/system-stacks/mesos/rancher-compose.yml]: floor(HOST_FAILURE_TOLERANCE * 2) + 1

File: resources/content/config-content/system-stacks/mesos/rancher-compose.yml [hunk 3]
```diff
@@ -7,6 +8,7 @@ zookeeper:
     healthy_threshold: 2
     response_timeout: 5000
 mesos-master:
+  scale: 3
```
Comment (LLParse) [inline resources/content/config-content/system-stacks/mesos/rancher-compose.yml]: floor(HOST_FAILURE_TOLERANCE * 2) + 1

File: resources/content/config-content/system-stacks/mesos/docker-compose.yml [hunk 4]
```diff
@@ -1,15 +1,27 @@
 zookeeper:
-  image: rawmind/rancher-zk:3.4.6-1
-  volumes:
-  - /opt/zk/data
+  image: rancher/zookeeper:3.4.8
   environment:
-    JVMFLAGS: -Xmx512m -Xms512m
-    CONFD_INTERVAL: 0
+    ZK_ENSEMBLE_SIZE: 3
```
Comment (LLParse) [inline resources/content/config-content/system-stacks/mesos/docker-compose.yml]: floor(HOST_FAILURE_TOLERANCE * 2) + 1

**Comment by cloudnautique** [comment]
LGTM

**Comment by cloudnautique** [comment]
OK, LGTM

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/4850

**Pull request merged**